### ISSUE TITLE
aws-rds: db:cluster unable to pass all props to instances

### ISSUE THREAD

**Issue by tehzwen** [aws/aws-cdk#20534]
### Describe the bug

You cannot specify `storageEncrypted` on the the generated instances of an rds cluster. The cluster can have its storage encrypted but some hackiness is required for the children instances to have said value

### Expected Behavior

Would expect that when the `storageEncrypted` value is passed to the cluster that any instance generated by that cluster would inherit that value or allow me to override it in a nice way.

### Current Behavior

```ts
   const cluster = new rds.DatabaseCluster(this, 'cluster', {
      engine: rds.DatabaseClusterEngine.auroraPostgres({
        version: rds.AuroraPostgresEngineVersion.VER_10_11
      }),
      credentials: rds.Credentials.fromSecret(databaseCredentialsSecret),
      instances: 1,
      instanceProps: {
        vpc,
      },
      storageEncrypted: true
    });
```
Will generate a cluster with a single instance that does not have the `storageEncrypted` value attached to it in the resulting cloudformation.

### Reproduction Steps

Create a cluster as shown above and a simple `cdk synth`, look in `cdk.out` at the resulting template and any db instance will not have the `storageEncrypted` value

### Possible Solution

Currently solvable with something like the following:
```ts
    cluster.node.children.forEach((c) => {
      if (c instanceof(rds.CfnDBInstance)) {
        let temp = c as rds.CfnDBInstance;
        temp.storageEncrypted = true;
      }
    });
```
However this is a bit hacky and not ideal. 

Allowing `0` instances then `addInstance(instance)` would solve this or allow instance props to contain all the fields that a normal instance has.

### CDK CLI Version

2.24.1

### Node.js Version

16.14.2

### OS

MacOS 12.4

### Language

Typescript

**Comment by tehzwen** [comment]
I can check for you, the issue we were having was that we had an SCP that disallowed `CreateDBInstance` calls without the encrypt field hence the work around was necessary

**Comment by tehzwen** [comment]
@corymhall so it looks like the child instance does in fact get the `storageEncrypted` value set once it is spun up. The issue we were having was that the initial `rds:CreateDBInstance` call does not contain it and our service control policies for db encryption block said api call if it does not contain it. The encryption value must be set/updated after the fact? A bit odd but glad it still gets applied

**Comment by corymhall** [comment]
Yeah this is a tricky one. You can pass the encryption config as part of the `CreateDbInstance` call, but if that database is part of a cluster it doesn't actually do anything. If the database instance is part of a cluster, the instance will inherit whatever encryption is set on the cluster. 

You might be able to update your SCPs to say that `CreateDBInstance` must have encryption _unless_ the cluster identifier is passed.

**Comment by tehzwen** [comment]
Yeah thanks for the rubber ducking, worse comes to worse the hacky (cosmetic) for loop will serve as a way to satisfy our SCP. 

Thanks again for the insight/help.
